CONTRIBUTING.md

```diff
@@ -58,3 +58,25 @@ import Foo as F
 import Qux (Q(..), q)
 import qualified A
 ```
+
+## Development Workflow
+
+Here we outline a general workflow new contributors could adopt for a nice incremental development experience.
+
+The root folder consists of two stack-*.yaml config files.
+
+1. stack-night.yaml - Config file defines all the settings to build and test any changes.
+2. stack-bench.yaml - Used to run benchmark tests.
+
+Therefore, a ghci development experience would be:
+
+1. `stack ghci --stack-yaml stack-nightly.yaml --test`
+2. :r to reload
+3. `main` to run all tests, or `:main --pattern Foo` to run specific tests matched by pattern.
+
+Some discussion on this topic can also be found here [here](https://github.com/bos/aeson/issues/733)
+
+Of course before submitting a PR, the following steps are recommended:
+
+1. Run the entire suite of tests and benchmarks.
+2. Run hlint on the source folders.
```

Comment:
I have not been able to run `make lint` locally. I ended up doing `hlint Data/*` for my most recent changes. 

 Here is the error:
```
$ make lint     
./run-hlint.sh --cpp-include include/ --cpp-file travis/cabal_macros.h Data/ attoparsec-iso8601/ benchmarks/ examples/ ffi/ pure/ tests/
./run-hlint.sh: line 6: ./hlint.sh: No such file or directory
Makefile:2: recipe for target 'lint' failed
make: *** [lint] Error 127
```

Looks like the run-hlint.sh is missing.

Reply:
I guess you don't have `hlint` on your PATH? 

Unless you pass all the args that are in `Makefile` you'll most likely get warnings and maybe not full linting. Invocation is a bit convoluted so that's why I added this file.

run-hlint.sh is a bit hacky because of travis, I forget exactly why I ended up doing it this way. Maybe it'd be nicer to add a `hlint.sh` with `hlint $@` to git, the travis job overwrites the file anyway.